pydantic_ai_slim/pydantic_ai/fastapi/agent_router.py

```diff
@@ -0,0 +1,165 @@
+
+
+class AgentAPIRouter(APIRouter):
+    """FastAPI Router for Pydantic Agent."""
+
+    def __init__(
+        self,
+        agent_registry: AgentRegistry,
+        disable_response_api: bool = False,
+        disable_completions_api: bool = False,
+        *args: tuple[Any],
+        **kwargs: tuple[Any],
+    ):
+        super().__init__(*args, **kwargs)
+        self.registry = agent_registry
+        self.responses_api = AgentResponsesAPI(self.registry)
+        self.completions_api = AgentChatCompletionsAPI(self.registry)
+        self.models_api = AgentModelsAPI(self.registry)
+        self.enable_responses_api = not disable_response_api
+        self.enable_completions_api = not disable_completions_api
+
+        # Registers OpenAI/v1 API routes
+        self._register_routes()
+
+    def _register_routes(self) -> None:  # noqa: C901
+        if self.enable_completions_api:
+
+            @self.post(
+                '/v1/chat/completions',
+                response_model=ChatCompletion,
+            )
+            async def chat_completions(  # type: ignore
+                request: ChatCompletionRequest,
+            ) -> ChatCompletion | StreamingResponse:
+                if not request.messages:
+                    raise HTTPException(
+                        status_code=400,
+                        detail=ErrorResponse(
+                            error=ErrorObject(
+                                type='invalid_request_error',
+                                message='Messages cannot be empty',
+                            ),
+                        ).model_dump(),
+                    )
+                try:
+                    if getattr(request, 'stream', False):
+                        return StreamingResponse(
+                            self.completions_api.create_streaming_completion(request),
+                            media_type='text/event-stream',
+                            headers={
+                                'Cache-Control': 'no-cache',
+                                'Connection': 'keep-alive',
+                                'Content-Type': 'text/plain; charset=utf-8',
+                            },
+                        )
+                    else:
+                        return await self.completions_api.create_completion(request)
+                except Exception as e:
+                    logger.error(f'Error in chat completion: {e}', exc_info=True)
+                    raise HTTPException(
+                        status_code=500,
+                        detail=ErrorResponse(
+                            error=ErrorObject(
+                                type='internal_server_error',
+                                message=str(e),
+                            ),
+                        ).model_dump(),
+                    )
+
+        if self.enable_responses_api:
+
+            @self.post(
+                '/v1/responses',
+                response_model=Response,
```

Comment:
We should rename the import above to `OpenAIResponse` or something like that, `Response` in the context of a web framework is already used.